examples/scripts/project/_common/fork_project.py

```diff
@@ -0,0 +1,126 @@
+
+import tamr_toolbox as tbox
+from tamr_toolbox import utils as tbu
+from tamr_toolbox.utils.client import Client
+from tamr_toolbox.utils.operation import Operation
+import argparse
+import os
+from pathlib import Path
+
+
+def export_from_tamr(client: Client, *, project_name: str, export_path: str,) -> Operation:
+    LOGGER.info(f"Project {project_name} export from Tamr initializing...")
+    project = client.projects.by_name(project_name)
+    if not export_path:
+        export_path = os.path.join(
+            Path(__file__).resolve().parent, f"project-movement/{project_name}"
+        )
+    return tbox.project.export_artifacts(project=project, artifact_directory_path=export_path)
+
+
+def import_to_tamr(
+    client: Client,
+    *,
+    zipfile_path: str,
+    new_project_name: str,
+    new_ud_name: str,
+    overwrite_existing: bool = False,
+) -> Operation:
+    LOGGER.info(f"Project {new_project_name} import to Tamr initializing...")
+    return tbox.project.import_artifacts(
+        tamr_client=client,
+        project_artifact_path=zipfile_path,
+        new_project_name=new_project_name,
+        new_unified_dataset_name=new_ud_name,
+        overwrite_existing=overwrite_existing,
+    )
+
+
+def main():
+    global LOGGER
+    # parse args
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--name", default=None, required=True, help="raise to specify the name of target project",
+    )
+    parser.add_argument(
+        "--exportPath",
+        default=None,
+        required=True,
+        help="raise to specify the path to export directory",
+    )
+    parser.add_argument(
+        "--newUDName",
+        default=None,
+        help="raise to explicitly specify the name of forked project unified dataset (optional)",
+    )
+    parser.add_argument(
+        "--overwrite",
+        default=False,
+        action="store_true",
+        help="if raised will replace a project with specified target name (if one already exists)",
+    )
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "--newName", default=None, help="raise to explicitly specify the name of the new project",
+    )
+    group.add_argument(
+        "--postfix",
+        default=None,
+        help="raise to imply the name of the new project by postfixing the original project name",
+    )
+    opts = parser.parse_args()
+
+    # creating the logger object:
+    logging_dir = "."
+    LOGGER = tbu.logger.create(__file__, log_directory=logging_dir)
+    # Let Tamr Toolbox itself also contribute to the log
+    tbu.logger.enable_toolbox_logging(log_directory=logging_dir, log_to_terminal=False)
+    # Configure the logs from imported packages
+    tbu.logger.enable_package_logging(
+        "tamr_unify_client", log_directory=logging_dir, log_to_terminal=False
+    )
+
+    # load config file and create tamr client
+    conf_dir = "."
+    config = tbu.config.from_yaml(conf_dir)
+
```

Comment:
Can move all of the argparse, logger setup, and config loading into the `if __name__ == "__main__":` clause?  That way, you don't need to make LOGGER a global, and you separate the command-line setup from the more interesting logic.

See [run_mastering_simple.py](https://github.com/Datatamer/tamr-toolbox/blob/ca0b9e4e09483d93d6312e5f0e87207c11b65cd3/examples/scripts/project/mastering/run_mastering_simple.py) for a good example of this.

Reply:
Absolutely - great suggestion, Sam. I am on it.

Reply:
After moving all of this, you will need to pass `opts` through `main()`, but `LOGGER` and `CONFIG` (good to capitalize it now) should be accessible throughout the script.